modules/ding_user/ding_user.module

```diff
@@ -386,13 +387,8 @@ function ding_user_form_user_admin_settings_alter(&$form, &$form_state, $form_id
   );
 
   $form['ding_user']['ding_user_pincode_length'] = array(
-    '#type' => 'radios',
+    '#type' => 'textfield',
     '#title' => t('Pin code length'),
-    '#options' => array(
-      4 => t('Max @number chars', array('@number' => 4)),
-      5 => t('Max @number chars', array('@number' => 5)),
-      6 => t('Max @number chars', array('@number' => 6)),
-    ),
```

Comment:
Please add [`element_validate_integer_positive`](https://api.drupal.org/api/drupal/includes!form.inc/function/element_validate_integer_positive/7.x) to ensure we have valid data.